languagetool-core/src/main/java/org/languagetool/rules/GenericUnpairedQuotesRule.java

```diff
@@ -0,0 +1,261 @@
+/* LanguageTool, a natural language style checker 
+ * Copyright (C) 2009 Daniel Naber (http://www.danielnaber.de)
+ * 
+ * This library is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this library; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+
+package org.languagetool.rules;
+
+import org.languagetool.AnalyzedSentence;
+import org.languagetool.AnalyzedTokenReadings;
+
+import java.text.MessageFormat;
+import java.util.*;
+import java.util.regex.Pattern;
+
+/**
+ * Rule that finds unpaired quotes, brackets etc.
+ * 
+ * @author Fred Kruse
+ * @since 6.4
+ */
+public class GenericUnpairedQuotesRule extends TextLevelRule {
```

Comment:
How much of this is duplicated from `GenericUnpairedBracketsRule`? We should avoid code duplication.

Reply:
> How much of this is duplicated from `GenericUnpairedBracketsRule`? We should avoid code duplication.

I started with a duplicate of the GenericUnpairedBracketsRule, but I had to change nearly everything. So there is no relevant duplicate code.

Reply:
Something else: This change makes my other two pull requests unnecessary. How can I cancel it?

Reply:
> How can I cancel it?

There's a "Close pull request" button at the bottom of e.g. https://github.com/languagetool-org/languagetool/pull/10129

Reply:
@danielnaber 
Hi Daniel, I would like to test the new rules against the Wikipedia dataset, but the self-compiled languagetool-wikipedia-6.4-SNAPSHOT.jar doesn't work (Error: java.lang.NoClassDefFoundError: org/languagetool/tools/JnaTools). The new rules are not included in the nightly. Please give me a hint.

Reply:
`languagetool-wikipedia-6.4-SNAPSHOT.jar` doesn't contain dependencies, but it will find them if you put it into the LT root directory and call it from there (the directory which has a `libs` folder with all the `.jar` files in it).